### ISSUE TITLE
[CI] HistoryIntegrationTests.testThatHistoryContainsStatus failures

### ISSUE THREAD

**Issue by tvernum** [elastic/elasticsearch#38693]
Test fails reproducibly.
CI failed on Java 12, but it reproduces on Java 11 for me.

From: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=openjdk12,nodes=immutable&&linux&&docker/235/console
```
05:50:09 FAILURE 1.44s J3 | HistoryIntegrationTests.testThatHistoryContainsStatus <<< FAILURES!
05:50:09    > Throwable #1: java.lang.AssertionError: 
05:50:09    > Expected: is "2019-02-11T05:50:07.998Z"
05:50:09    >      but: was "2019-02-11T05:50:07.998586Z"
05:50:09    > 	at __randomizedtesting.SeedInfo.seed([F30FA9CCD903E4B5:FE0ACADB0B20E6F7]:0)
05:50:09    > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)

./gradlew :x-pack:plugin:watcher:unitTest \
  -Dtests.seed=F30FA9CCD903E4B5 \
  -Dtests.class=org.elasticsearch.xpack.watcher.test.integration.HistoryIntegrationTests \
  -Dtests.method="testThatHistoryContainsStatus" \
  -Dtests.security.manager=true \
  -Dtests.locale=it-IT \
  -Dtests.timezone=Asia/Shanghai \
  -Dcompiler.java=11 \
  -Druntime.java=12
```

Relates: #38505, #27330

**Comment by pgomulka** [comment]
should be fixed by https://github.com/elastic/elasticsearch/pull/38626

**Comment by pgomulka** [comment]
but might as well change this to use a formatter that it uses during toXContent (as per comment https://github.com/elastic/elasticsearch/issues/38619#issuecomment-461809234)
```
XContentElasticsearchExtension.DEFAULT_FORMATTER
```

**Comment by pgomulka** [comment]
closing as it is fixed in  #38626
